edi-module/component_group_reader.bal

```diff
@@ -0,0 +1,64 @@
+class ComponentGroupReader {
+
+    SubcomponentGroupReader subcomponentGroupReader = new();
+
+    function read(string compositeText, EDIMapping mapping, EDIFieldMapping emap)
+                returns EDIComponentGroup|error? {
+        if compositeText.trim().length() == 0 {
+            // Composite value is not provided. Return null, which will cause this field to be not included.
+            return null;
```

Comment:
```suggestion
            return ();
```

Reply:
`null` is used only in json context and to represent ballerina nil value use `()` always.